File: src/kompas_builder/kompas_app.py
import pythoncom
from win32com.client import DispatchEx

class KompasApp:
    def __init__(self):
        pythoncom.CoInitialize()
        self.app5 = DispatchEx("Kompas.Application.5")
        self.app5.Visible = False

    def get_param_struct(self, code):
        return self.app5.GetParamStruct(code)

    def quit(self):
        pass
        try:
            self.app5.Quit()
        except Exception:
            pass
        finally:
            del self.app5
            pythoncom.CoUninitialize()

File: cpp_tool/src/KompasApp.cpp
#include "KompasApp.h"

KompasApp::KompasApp() {
  // Инициализация COM
  CoInitialize(nullptr);

  // Создаём COM-объект "Kompas.Application.5"
  kompas.CreateInstance(L"KOMPAS.Application.5");
  kompas->Visible = VARIANT_FALSE; // скрыть GUI
}

KompasApp::~KompasApp() {
  kompas->Quit();
  CoUninitialize();
}

DocumentParamPtr KompasApp::getParamStruct(long code) {
  // GetParamStruct возвращает IDispatch*, приводим к DocumentParamPtr
  return (DocumentParamPtr)kompas->GetParamStruct(code);
}

Document2DPtr KompasApp::getDocument2D() { return kompas->Document2D(); }
